### PR TITLE
test(storage): store the full Status in ThroughputResult

### PR DESCRIPTION
**Pull request by mr-salty**
This may give us more information if googleapis#5202 recurs.

coryan: one thing that changed is the PrintAsCsv() format, is that going
to mess up any of your analysis? Should I change it?

INSERT,3145728,2097152,4194304,1,0,GRPC,234000,345000,OOR-status-message [OUT_OF_RANGE]

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5203)

### CODE REVIEW

**Comment by coryan** [comment]
> coryan: 

nit: `@coryan` is the way to mention folks on GitHub.

> one thing that changed is the PrintAsCsv() format, is that going to mess up any of your analysis? Should I change it?

I think error messages could contain `,` and `\n` characters (I have actually seen the latter), and that would certainly mess by analysis.  Can we convert those to `;` or some character like it?

**Review by coryan — COMMENTED** [review]
Maybe change the `ostream<<` operator to replace newlines and commas by some other character?

**Review by devbww — COMMENTED** [review]
For what it's worth, these sound like things that a `PrintAsCsv()` function should handle (e.g., by escaping `,` and '\n' chars).

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5203#-:-MJ8LDnJEE3lhi_GCdXq:b-8k2080)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by coryan — APPROVED** [review]

**Review by mr-salty — COMMENTED** [review]
PTAL - I quoted the string (AFAICT `status` is the only field affected).

Note that [rfc 4180](https://tools.ietf.org/html/rfc4180#section-2) does *not* say you have to quote if the string contains spaces - just comma, double-quote, or CRLF (2.6). Although, I chose to escape CR|LF, and could be convinced to do it for spaces as well.  It's always *permissible* to quote the string. per 2.5.

The character set is also supposed to be limited, although I didn't do anything to strip out other characters:    `TEXTDATA =  %x20-21 / %x23-2B / %x2D-7E`

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5203#-:-MJ92EXd54BiigbZzYYg:b4dx7bn)* status: 4 of 6 files reviewed, all discussions resolved (waiting on @devbww)

File: google/cloud/storage/benchmarks/throughput_result.cc [hunk 1]
```diff
@@ -13,16 +13,36 @@
...
 namespace cloud {
 namespace storage_benchmarks {
 
+namespace {
+template <typename T>
+std::string QuoteCsv(T const& element) {
+  std::ostringstream os;
+  os << element;
+  std::string result = os.str();
+  if (result.find_first_of(",\r\n") == std::string::npos) {
```
Comment (devbww) [inline google/cloud/storage/benchmarks/throughput_result.cc:31]: Double quote?

**Comment by mr-salty** [comment]
___
*[google/cloud/storage/benchmarks/throughput_result.cc, line 31 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5203#-MJ91xXsazuyvwcWOzUF-r2-31:-MJ9gBSCAGbFfP_o4mK_:b7z55tb) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/00504971e3a62012972de7db80406e2eb21613bc/google/cloud/storage/benchmarks/throughput_result.cc#L31)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Double quote?
</blockquote></details>

good catch, I knew I should have done tests for each case separately (and now I have).

**Review by devbww — APPROVED** [review]

**Pull request merged**